config/webhook/webhook.go

```diff
@@ -33,14 +37,54 @@ func main() {
 }
 
 type EvictionHandler struct {
+	Client  client.Client
 	decoder *admission.Decoder
 }
 
 func (e *EvictionHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
 	logger.Info("Received eviction request", "namespace", req.Namespace, "name", req.Name)
 
 	// Log eviction request
-	fmt.Printf("Eviction requested for Pod %s/%s\n", req.Namespace, req.Name)
+	evictionLog := myappsv1.EvictionLog{
+		PodName:      req.Name,
+		EvictionTime: time.Now().Format(time.RFC3339),
+	}
+
+	// List all PDBWatchers
+	pdbWatcherList := &myappsv1.PDBWatcherList{}
+	err := e.Client.List(ctx, pdbWatcherList, &client.ListOptions{Namespace: req.Namespace})
+	if err != nil {
+		logger.Error(err, "Unable to list PDBWatchers")
+		return admission.Errored(http.StatusInternalServerError, err) // don't want to block eviction
+	}
+
+	// Find the applicable PDBWatcher
+	var applicablePDBWatcher *myappsv1.PDBWatcher
+	for _, pdbWatcher := range pdbWatcherList.Items {
+		// Add your logic to identify the applicable PDBWatcher
```

Comment:
what's your plan here? do a get on each watchers pdb and then see if that pdb selects the evicted pod? 
what's stopping you from doiing that now.

Reply:
Also might be worth considering unittesting which will e a bit eaiser if you have a function that takes an eviction a fake k8s client.